app/src/main/kotlin/com/gdgnantes/devfest/android/app/PreferencesManager.kt

```diff
@@ -73,3 +75,14 @@ private fun SharedPreferences.string(name: String, defaultValue: String? = null)
     }
 }
 
+private fun SharedPreferences.stringSet(name: String, defaultValue: Set<String> = emptySet())
+        = object : ReadWriteProperty<Any, Set<String>> {
+    override fun getValue(thisRef: Any, property: KProperty<*>): Set<String> {
+        return getStringSet(name, defaultValue)
```

Comment:
The thing is the returned type is a `Set` so it's already immutable from an API point of view.